test/index.test.js

```diff
@@ -32,299 +35,386 @@ Object.defineProperty(loaderContextMock, "options", {
+                                        {
+                                            loader: pathToSassLoader, options: {
+                                                implementation,
+                                                sourceMap: true
+                                            }
+                                        }
+                                    ]
+                                }]
+                            }
+                        }, (webpackErr, stats) => {
+                            const err = webpackErr ||
+                                  (stats.hasErrors() && stats.compilation.errors[0]) ||
+                                  (stats.hasWarnings() && stats.compilation.warnings[0]);
+
+                            if (err) {
+                                reject(err);
+                            } else {
+                                resolve();
+                            }
+                        });
+                    });
+                }
 
-        it("should compile without errors", () => buildWithSourceMaps());
-        it("should produce a valid source map", () => {
-            const cwdGetter = process.cwd;
-            const fakeCwd = path.join(__dirname, "scss");
+                it("should compile without errors", () => buildWithSourceMaps());
+                it("should produce a valid source map", () => {
+                    const cwdGetter = process.cwd;
+                    const fakeCwd = path.join(__dirname, "scss");
 
-            process.cwd = function () {
-                return fakeCwd;
-            };
+                    process.cwd = function () {
+                        return fakeCwd;
+                    };
 
-            return buildWithSourceMaps()
-                .then(() => {
-                    const sourceMap = testLoader.sourceMap;
+                    return buildWithSourceMaps()
+                        .then(() => {
+                            const sourceMap = testLoader.sourceMap;
 
-                    sourceMap.should.not.have.property("file");
-                    sourceMap.should.have.property("sourceRoot", fakeCwd);
-                    // This number needs to be updated if imports.scss or any dependency of that changes
-                    sourceMap.sources.should.have.length(13);
-                    sourceMap.sources.forEach(sourcePath =>
-                        fs.existsSync(path.resolve(sourceMap.sourceRoot, sourcePath))
-                    );
+                            sourceMap.should.not.have.property("file");
+                            sourceMap.should.have.property("sourceRoot", fakeCwd);
+                            // This number needs to be updated if imports.scss or any dependency of that changes.
+                            // Node Sass includes a duplicate entry, Dart Sass does not.
+                            sourceMap.sources.should.have.length(implementation == nodeSass ? 12 : 11);
```

Comment:
We can get rid of the duplicated dependency. Makes this test easier. Which dependency is duplicated?